Update MQTT update platform documentation (#24723)
home-assistant/home-assistant.github.io: source/_integrations/update.mqtt.markdown

  required: false
  type: integer
  default: 0
release_summary:

>   description: Summary of the release notes or changelog. This is not suitable for long changelogs but merely suitable for a short excerpt update description of max 255 characters.

Reviewer: ```suggestion
  description: Summary of the release notes or changelog. This is suitable a brief update description of max 255 characters.
```

  required: false
  type: string
release_url:
  description: URL to the full release notes of the latest version available.
  required: false
  type: string
retain:
  description: If the published message should have the retain flag on or not.
  required: false
  type: boolean
  default: false
state_topic:

>   description: The MQTT topic subscribed to receive state updates. The state update may be either JSON or a simple string with `installed_version` value. When a JSON payload is detected, the state value of the JSON payload could supply the `installed_version`, `latest_version`, `title`, `release_summary`, `release_url` or `entity_picture` values.

Reviewer: ```suggestion
  description: The MQTT topic subscribed to receive state updates. The state update may be either JSON or a simple string with `installed_version` value. When a JSON payload is detected, the state value of the JSON payload should supply the `installed_version` and can optional supply:  `latest_version`, `title`, `release_summary`, `release_url` or an `entity_picture` URL.
```

  required: false
  type: string
title:

>   description: Title of the software. This helps to differentiate between the device or entity name versus the title of the software installed.

Reviewer: ```suggestion
  description: Title of the software, or firmware update. This helps to differentiate between the device or entity name versus the title of the software installed.
```

  required: false
  type: string
unique_id:
  description: An ID that uniquely identifies this Select. If two Selects have the same unique ID Home Assistant will raise an exception.
  required: false
  type: string
value_template:

>   description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the installed version value. Alternatively `value_template` can be used to render to a valid JSON payload."

Reviewer: ```suggestion
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the `installed_version` state value or to render to a valid JSON payload on from the payload received on `state_topic`."
```
